Add ~/.local/share/omarchy/bin to systemwide PATH

ryanrhughes opened the pull request:
Adds to the system path via UWSM. Also tested adding via proflile.d but that didn't get picked up everywhere, particularly by Walker whereas this does.

<img width="1220" height="966" alt="screenshot-2025-08-10_01-43-23" src="https://github.com/user-attachments/assets/0a609078-94c1-4235-bdaf-8bc111ab0031" />

<img width="1428" height="546" alt="image" src="https://github.com/user-attachments/assets/9423a17a-b4b4-4248-9aa3-54c710a130db" />

Fixes #557

manuel1618 commented on bin/omarchy-menu:
```diff
@@ -52,7 +50,7 @@ install_font() {
 show_learn_menu() {
   case $(menu "Learn" "  Keybindings\n  Omarchy\n  Hyprland\n󰣇  Arch\n  Neovim\n󱆃  Bash") in
-  *Keybindings*) $OMARCHY_BIN_PATH/omarchy-menu-keybindings ;;
+  *Keybindings*) alacritty --class Omarchy -e bash -c "~/.local/share/omarchy/bin/omarchy-show-logo; echo $PATH; ~/.local/share/omarchy/bin/omarchy-show-done;" ;;
```
I'm curious to know why you replaced the keybindings bin with a custom command?

ryanrhughes replied:
Accidental debug inclusion. Thanks for catching.

dhh commented on migrations/1754680295.sh:
```diff
@@ -0,0 +1,3 @@
+echo "Add UWSM env"
+mkdir -p "$HOME/.config/uwsm/"
+omarchy-refresh-config uwsm/env
```
Do we need to restart the uwsm session for this to get picked up? Could do that with a gum confirm, if needed.

ryanrhughes replied:
Sure do. Pushed that up.

dhh commented:
Now question is if this happens as part of a larger upgrade that has later migrations? I guess you'll be able to run those migrations with an update later, but might need the system in a half-finished state along the way. Hmm!

dhh commented:
Question still is, though, that if we change this now, then later migrations that depend on the PATH, might fail in the same upgrade session? Wonder if there's any clutch we can do for the PATH munging in the omarchy-menu or omarchy-update script to make it work either way?

ryanrhughes commented:
I thought about that a touch but couldn't come up with a viable scenario this could happen. 

If you run update and get the updated version of the menu where the path var is gone, you will also have had to run the migration for this already.

Migration processes execute in the user env where we already have omarchy/bin as a part of the path and is done via sourcing the files which should keep us in the same terminal where the `$PATH` was set before we removed it from the defaults. However, I'll confirm this with some fake test migrations while working in the refresh / restart drop in.

The only scenarios I could think of that could arise aside from this are:
- If you did a manual `git pull` on `dev` but at that point, there are likely other problems you'll encounter as well not running migrations.  Entirely possible I'm overlooking an obvious situation here though.
- If you deny restarting, but then try to use the menu at which point; we warned you. 🤷‍♂

Is there another you can think of playing out that I'm not thinking about I can test as well?

ryanrhughes commented:
Added a simple state set / clear script with some checks at the end of the update. We detect in order from most aggressive to least so if we need to restart, who cares about the relaunch. Restart clears both states.

Now in future migrations, we can just drop `omarchy-state set <restart-required | relaunch-required>` and be good

I added a migration downstream too to test my theory about paths and it did what I expected. Because the `$PATH` was set when the terminal was initiated with omarchy/bin as a part of the path, it maintained it for the subsequent files to run.

<img width="2786" height="376" alt="image" src="https://github.com/user-attachments/assets/53ae3ce5-4b00-4996-ad7b-bfdce26d1889" />

dhh commented:
One thing I just realized is that this won't ask for a restart on the first run, because it'll be running an old version of install.sh, which doesn't have that logic.

dhh commented:
We'll still possibly have a problem on the first run. But I'll test that.
